wcomponents-theme/src/main/js/wc/file/util.js

```diff
@@ -0,0 +1,135 @@
+define(["wc/dom/uid", "wc/file/getMimeType", "wc/config"], function (uid, getMimeType, wcconfig) {
+	"use strict";
+
+	var instance = new FileUtil(),
+
+		/**
+		 * Used when checking the newly created file is named with the correct extension.
+		 * If it does not already match any in the array then the extension at index zero will be appended.
+		 **/
+		mimeToExt = {
+			"image/jpeg": ["jpeg", "jpg"],
+			"image/bmp": ["bmp"],
+			"image/png": ["png"],
+			"image/gif": ["gif"],
+			"image/x-icon": ["ico"],
+			"image/webp": ["webp"],
+			"image/svg+xml": ["svg"],
+			"image/tiff": ["tif", "tiff"],
+			"text/plain": ["bas", "c", "h", "txt", "log", "bat"],
```

Comment:
wouldn't "txt" be the most logical first entry for "text/plain"?

Reply:
Agree, interestingly even [mimemap.json](https://github.com/BorderTech/wcomponents/blob/3e8e1d947be645e0747274e80b80e2be608fdcfd/wcomponents-theme/src/main/resource/mimemap.json#L10) has `bas` as first entry.